.github/workflows/test.yml

```diff
@@ -65,8 +65,10 @@ jobs:
       - uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha || github.sha }}
-          # check out full history
-          fetch-depth: 0
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
```

Comment:
this says 20 but we use 18 in the yarn-install step - should we consolidate? also in the yarn-install we setup before checking out, i don't mind which order we do it but would prefer consistency

Reply:
I used 20 because that's what is set in the `.nvmrc` file. Maybe we could start sourcing the version from that file as it directs our dev machine workflow. I'll create a separate PR for this.

Reply:
can you also update the Dockerfile as a drive-by with that PR please 🙏 https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/6101959f7799faa1e92056fb4afb39e3edf8d867/Dockerfile#L1

Reply:
There's a built-in way of reading the version from `.nvmrc` -- I'm testing if it works.